test/1.4/expressions/T_discard_ref.dml

```diff
@@ -5,6 +5,8 @@
 dml 1.4;
 device test;
 
+/// DMLC-FLAG --no-compat=discard_ref_shadowing
```

Comment:
Without this, the test passed in t126 before I added objident_or_underscore.

However, I suppose the error would still have been captured in the 7 build, so one may argue that adding the flag reduces test coverage in CI (the file is never compiled *without* the flag). So perhaps this commit should be dropped.